src/pages/Dashboard/Modals/BorrowRepay/Repay/index.spec.tsx

```diff
@@ -54,32 +53,238 @@ describe('pages/Dashboard/BorrowRepayModal/Repay', () => {
+      target: { value: incorrectValueTokens },
     });
 
-    await waitFor(() => getByText(en.borrowRepayModal.repay.submitButton));
-    expect(getByText(en.borrowRepayModal.repay.submitButton).closest('button')).not.toHaveAttribute(
-      'disabled',
+    // Check submit button is disabled
+    await waitFor(() => getByText(en.borrowRepayModal.repay.submitButtonDisabled));
+    expect(
+      getByText(en.borrowRepayModal.repay.submitButtonDisabled).closest('button'),
+    ).toHaveAttribute('disabled');
+  });
+
+  it('disables submit button if an amount entered in input is higher than token wallet balance', async () => {
+    const { getByText, getByTestId } = renderComponent(
+      <AuthContext.Provider
+        value={{
+          login: jest.fn(),
+          logOut: jest.fn(),
+          openAuthModal: jest.fn(),
+          closeAuthModal: jest.fn(),
+          account: {
+            address: fakeAccountAddress,
+          },
+        }}
+      >
+        <Repay asset={fakeAsset} onClose={noop} isXvsEnabled />
+      </AuthContext.Provider>,
     );
+    await waitFor(() => getByText(en.borrowRepayModal.repay.submitButtonDisabled));
+
+    expect(
+      getByText(en.borrowRepayModal.repay.submitButtonDisabled).closest('button'),
+    ).toHaveAttribute('disabled');
+
+    const incorrectValueTokens = fakeAsset.borrowBalance.plus(1).toFixed();
 
-    // Enter amount higher than maximum borrow limit in input
+    // Enter amount in input
     fireEvent.change(getByTestId('token-text-field'), {
-      target: { value: fakeAsset.borrowBalance.plus(1).toFixed() },
+      target: { value: incorrectValueTokens },
     });
+
+    // Check submit button is disabled
     await waitFor(() => getByText(en.borrowRepayModal.repay.submitButtonDisabled));
     expect(
       getByText(en.borrowRepayModal.repay.submitButtonDisabled).closest('button'),
     ).toHaveAttribute('disabled');
   });
 
+  it('updates input value to token wallet balance when pressing on max button if token wallet balance is lower than token borrow balance', async () => {
+    const customFakeAsset: Asset = {
+      ...fakeAsset,
+      borrowBalance: new BigNumber(100),
+      walletBalance: new BigNumber(10),
+    };
+
+    const { getByText, getByTestId } = renderComponent(
+      <AuthContext.Provider
+        value={{
+          login: jest.fn(),
+          logOut: jest.fn(),
+          openAuthModal: jest.fn(),
+          closeAuthModal: jest.fn(),
+          account: {
+            address: fakeAccountAddress,
+          },
+        }}
+      >
+        <Repay asset={customFakeAsset} onClose={noop} isXvsEnabled />
+      </AuthContext.Provider>,
+    );
+    await waitFor(() => getByText(en.borrowRepayModal.repay.submitButtonDisabled));
+
+    // Check input is empty
+    const input = getByTestId('token-text-field') as HTMLInputElement;
+    expect(input.value).toBe('');
+
+    // Press on max button
+    fireEvent.click(getByText('MAX'));
```

Comment:
I have been importing translations here because if the wording changes the tests don't have to break

Reply:
Yeah that's a better option, I think I just missed this one. I'll update